src/test/java/opg/openml/webapplication/evaluator/TestDatasetEvaluator.java

```diff
@@ -0,0 +1,54 @@
+package opg.openml.webapplication.evaluator;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+
+import org.junit.Test;
+import org.openml.apiconnector.algorithms.Conversion;
+import org.openml.apiconnector.io.OpenmlConnector;
+import org.openml.apiconnector.settings.Constants;
+import org.openml.apiconnector.settings.Settings;
+import org.openml.apiconnector.xml.DataFeature;
+import org.openml.apiconnector.xml.DataQuality;
+import org.openml.apiconnector.xml.DataSetDescription;
+import org.openml.apiconnector.xstream.XstreamXmlMapping;
+import org.openml.webapplication.ProcessDataset;
+import org.openml.webapplication.testutils.DatasetFactory;
+
+import com.thoughtworks.xstream.XStream;
+
+import weka.core.Instances;
+
+public class TestDatasetEvaluator {
+	
+
+	private static final String testUrl = "https://test.openml.org/";
+	private static final String testKey = "d488d8afd93b32331cf6ea9d7003d4c3"; // admin key
+	private static final OpenmlConnector clientAdmin = new OpenmlConnector(testUrl, testKey);
+	private static final XStream xstream = XstreamXmlMapping.getInstance();
+	
+	@Test
+	public final void testActivateDataset() throws Exception {
+		Settings.CACHE_ALLOWED = false;
+		// first upload a dataset
+		DataSetDescription dsd = new DataSetDescription("testXor", "testXor", "arff", "class");
+		Instances dataset = DatasetFactory.getXORNominal();
+		File description = Conversion.stringToTempFile(xstream.toXML(dsd), "description", "xml");
+		File datasetFile = Conversion.stringToTempFile(dataset.toString(), "xor", "arff");
+		int did = clientAdmin.dataUpload(description, datasetFile).getId();
+		
+		new ProcessDataset(clientAdmin, did, null);
```

Comment:
Should this be something like clientAdmin.processDataset(did)? I'm not why we create an object that is not used later on (or I don't see how)?

Reply:
`clientAdmin` is the openml connector object. This is used several times to connect with OpenML, but does not have the capacity to evaluate datasets. 

it is true that it is weird that `ProcessDataset` has all its functionality in the constructor, and that is weird. We should change that at some point.